components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApiResource.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model;
+
+import java.util.List;
+
+/**
+ * API resource.
+ */
+public class ApiResource {
+
+    private static final long serialVersionUID = 2199048941051702943L;
+    private String id;
+    private Integer cursorKey;
+    private String name;
+    private String type;
+    private String identifier;
+    private String description;
+    private Integer tenantId;
+    private boolean requiresAuthorization;
+    private List<Scope> scopes;
+    private List<ApplicationBasicInfo> subscribedApplications;
+
+    public ApiResource() {
+    }
+
+    public ApiResource(String apiResourceId, String apiResourceName, String apiResourceIdentifier, String description,
+                       int apiResourceTenantId, String apiResourceType, boolean requiresAuthorization) {
+
+        this.id = apiResourceId;
+        this.name = apiResourceName;
+        this.identifier = apiResourceIdentifier;
+        this.description = description;
+        this.tenantId = apiResourceTenantId;
+        this.type = apiResourceType;
+        this.requiresAuthorization = requiresAuthorization;
+    }
+
+    public String getId() {
```

Comment:
missing newline after the method signature. check other places as well

Reply:
https://github.com/wso2/carbon-identity-framework/pull/4854/commits/8dd16447d6000412e3ff2dd1b2a9ed78249f2d13